.github/workflows/register-buildpack.yml

```diff
@@ -0,0 +1,260 @@
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Node Dependencies
+        working-directory: .github/workflows/scripts/validate
+        shell: bash
+        run: |
+          npm install
+      - id: detect_validate
+        name: Validate Buildpack Issue
+        uses: actions/github-script@v2
+        with:
+          result-encoding: string
+          script: |
+            const path = require('path')
+            const scripts = require(path.resolve('.github/workflows/scripts/validate/index.js'))
+            let result
+            try {
+              result = scripts.validateIssue({context})
+            } catch (error) {
+              console.error(error)
+              process.exit(1)
+            }
+            console.log(`::set-env name=BUILDPACK::${JSON.stringify(result)}`)
+
+      - uses: actions/github-script@v2
+        name: Retrieve Owners
+        with:
+          github-token: ${{ secrets.BOT_TOKEN }}
+          script: |
+            const buildpackInfo = JSON.parse(process.env.BUILDPACK)
+            let registryOwners = ''
+            try {
+              const {data} = await github.repos.getContent({
+                owner: process.env.GITHUB_OWNER,
+                path: `v1/${buildpackInfo.ns}.json`,
+                repo: process.env.NAMESPACE_REPO
+              })
+              const buff = new Buffer.from(data.content, 'base64')
+              registryOwners = buff.toString('utf-8')
+
+            } catch (error) {
+              if (error.status && error.status === 404) {
+                console.error('Creating file since it does not exist')
+                const content = {
+                  owners: [
+                    {
+                      id: context.payload.sender.id,
+                      type: 'github_user'
+                    }
+                  ]
+                };
+                const buff = Buffer.from(JSON.stringify(content), 'utf-8');
+                registryOwners = buff.toString('utf-8')
+
+                await github.repos.createOrUpdateFile({
+                  owner: process.env.GITHUB_OWNER,
+                  repo: process.env.NAMESPACE_REPO,
+                  path: `v1/${buildpackInfo.ns}.json`,
+                  message: 'initial commit',
+                  content: buff.toString('base64'),
+                  committer: {
+                    name: process.env.GITHUB_OWNER,
+                    email: 'longoria.public@gmail.com'
+                  },
+                  author: {
+                    name: process.env.GITHUB_OWNER,
+                    email: 'longoria.public@gmail.com'
+                  }
+                })
+              } else {
+                console.error(error)
+                process.exit(1)
+              }
+            }
+            console.log(`::set-env name=REGISTRY_OWNERS::${registryOwners.replace(/\r?\n|\r/g, " ")}`)
+
+      - uses: actions/github-script@v2
+        name: Authenticate
```

Comment:
Could you clarify the authentication flow? If you're the initial person who registered a buildpack, you are authenticated, but how/when does a `github_org` get added to the owners file?

Reply:
You are correct.  By default, if there is no `owner` defined in that namespace repo, the github user who submitted the issue becomes the owner.   For now, any changes to the owners will have to be gated via a PR to the namespace repo.  This may change down, but we (Joe and I) decided on this approach initially. 

Flow:
1) Check for owners in the nampespace
2) If no owners exist, create owners file with the issue author as a member of the owner [done]
3) if owners exist, examine the list owners for a match e.g. the issue author github id is in the list
4) if github user id isn't found in the owners list, check to see if there are any github organizations in the owner list
5) if the github user belongs to any of the github organizations in the list, the user is authorized [done]
6) if the github user is also not found in any of the github organziations, the user is NOT authorized [done]

Reply:
How does a github organization get added to the list?

Reply:
I don't believe that would be part of the spec but more an implementation detail of how a registry operator would like to manage access control. 

The best reference of this is in the RFC: https://github.com/buildpacks/rfcs/blob/main/text/0022-client-side-buildpack-registry.md#namespace-ownership